### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by dbartalino**
- Added default Express error handler
- Added next function for error handling in endpoints
- Added error handling for deleting a claim when there is no matching VIN, instead of returning "undefined"

### CODE REVIEW

File: middleware/lib/claim-client.js [hunk 1]
```diff
@@ -144,7 +144,12 @@ class ClaimClient {
    *
    * @param {string} vin - The VIN of the claim to delete.
    */
-  async deleteClaim(vin) {
+  async deleteClaim(vin, { vehicle = {} }) {
```
Comment (dbartalino) [inline middleware/lib/claim-client.js]: @ej3076 - If @dsifford agrees we should have this check for delete claim, this will require an update to send the VIN in the body so it can check if it exists/matches the VIN parameter, similar to editing a claim.

Comment (dsifford) [reply middleware/lib/claim-client.js]: I'm not sure I understand what you're describing here.

File: processors/src/claim-processor/handler.js [hunk 2]
```diff
@@ -81,6 +81,11 @@ class ClaimHandler extends TransactionHandler {
    */
   async deleteClaim({ data: claim }, state) {
     const { vin } = claim.vehicle;
+    const existingClaim = await state.getClaim(vin);
```
Comment (dsifford) [inline processors/src/claim-processor/handler.js]: Why is this necessary?

This exact same thing happens the way that it is currently written.

File: processors/src/claim-processor/handler.js [hunk 3]
```diff
@@ -95,7 +100,7 @@ class ClaimHandler extends TransactionHandler {
     const existingClaim = await state.getClaim(vin);
     if (!existingClaim) {
       logger.error(`Attempted to edit a non-existing claim for VIN: ${vin}.`);
-      throw new InvalidTransaction('Claim does not exist');
+      throw new InvalidTransaction('Cannot edit - claim does not exist');
```
Comment (dsifford) [inline processors/src/claim-processor/handler.js]: Why??????????

**Review by dsifford — CHANGES_REQUESTED** [review]

File: processors/src/claim-processor/handler.js (hunk 2 above)
Comment (dbartalino) [reply processors/src/claim-processor/handler.js]: Without this, trying to delete a nonexistent claim doesn't do anything (which is fine) but just returns "undefined", the same message that's returned when a claim is deleted.  This way we at least get an error message returned.

File: processors/src/claim-processor/handler.js (hunk 3 above)
Comment (dbartalino) [reply processors/src/claim-processor/handler.js]: Just differentiating between the invalid transaction messages for deleting and editing a claim.  I can change it back so they both return the same message if necessary.  Not a big deal.

File: middleware/lib/claim-client.js (hunk 1 above)
Comment (dbartalino) [reply middleware/lib/claim-client.js]: You can disregard this.  This is not needed - I was trying to mimic the edit claim errors too closely.

File: processors/src/claim-processor/handler.js (hunk 2 above)
Comment (dsifford) [reply processors/src/claim-processor/handler.js]: Why should it even be an error? Who cares if it doesn't exist. The same result occurs either way -> claim doesn't exist.

File: processors/src/claim-processor/handler.js (hunk 3 above)
Comment (dsifford) [reply processors/src/claim-processor/handler.js]: The differentiation isn't necessary because it's implicitly differentiated by whoever calls the api.

Comment (dsifford) [reply processors/src/claim-processor/handler.js]: When I call "edit claim" and get an error, it's obvious that the error happened in edit claim.

File: processors/src/claim-processor/handler.js (hunk 2 above)
Comment (dsifford) [reply processors/src/claim-processor/handler.js]: Also, most `DELETE` requests don't return a body. I'm not aware of a single api that does actually.

**Review by dsifford — APPROVED** [review]
